packages/react-jss/src/injectSheet.js

```diff
@@ -29,6 +29,9 @@ export default function injectSheet(styles: StylesOrCreator, options?: Options =
   return (InnerComponent: ComponentType<InnerProps> = NoRenderer) => {
     const Jss = createHoc(styles, InnerComponent, {...options, index})
 
+    // $FlowFixMe
+    Jss.InnerComponent = InnerComponent
```

Comment:
Mb if we place it as a static property in Jss component it won't complain?

Reply:
That won't fix it, a React Component type simply doesn't have a InnerComponent prop